src/tls.rs

```diff
@@ -7,14 +7,58 @@ use std::sync::Arc;
 use std::time::SystemTime;
 use tokio_rustls::rustls::{
     self,
-    client::{ServerCertVerified, ServerCertVerifier},
-    Certificate, PrivateKey, ServerName,
+    client::{verify_server_cert_signed_by_trust_anchor, ServerCertVerified, ServerCertVerifier},
+    server::ParsedCertificate,
+    Certificate, PrivateKey, RootCertStore, ServerName,
 };
 use tokio_rustls::TlsAcceptor;
 
 use crate::config::get_config;
 use crate::errors::Error;
 
+use arc_swap::ArcSwap;
+use log::{error, info};
+use once_cell::sync::Lazy;
+
+static ROOT_CERT_STORE: Lazy<ArcSwap<RootCertStore>> =
+    Lazy::new(|| ArcSwap::from_pointee(RootCertStore::empty()));
+
+/// Get a read-only instance of the configuration
+/// from anywhere in the app.
+/// ArcSwap makes this cheap and quick.
+pub fn get_root_cert_store() -> RootCertStore {
+    (*(*ROOT_CERT_STORE.load())).clone()
```

Comment:
A clone is expensive, I think you want to pass by reference here and drop that reference as quickly as possible after validating the certificate.

Reply:
Then please tell me how I can competently take a reference to this object?

Reply:
> Then please tell me how I can competently take a reference to this object?

I wrote that for a reason. You have a similar mechanism used in your entire code